homeassistant/components/screenlogic/binary_sensor.py

```diff
@@ -29,69 +30,70 @@ async def async_setup_entry(
     async_add_entities: AddEntitiesCallback,
 ) -> None:
     """Set up entry."""
-    entities = []
-    coordinator = hass.data[DOMAIN][config_entry.entry_id]
+    entities: list[ScreenLogicBinarySensorEntity] = []
+    coordinator: ScreenlogicDataUpdateCoordinator = hass.data[DOMAIN][
+        config_entry.entry_id
+    ]
+    gateway_data = coordinator.gateway_data
+    chemistry = gateway_data[SL_DATA.KEY_CHEMISTRY]
+    config = gateway_data[SL_DATA.KEY_CONFIG]
 
     # Generic binary sensor
-    entities.append(ScreenLogicBinarySensor(coordinator, "chem_alarm"))
+    entities.append(
+        ScreenLogicStatusBinarySensor(coordinator, "chem_alarm", CODE.STATUS_CHANGED)
+    )
 
     entities.extend(
         [
-            ScreenlogicConfigBinarySensor(coordinator, cfg_sensor)
-            for cfg_sensor in coordinator.data[SL_DATA.KEY_CONFIG]
```

Comment:
In a future PR : We could probably refactor this to use BinarySensorEntityDescription with a lambda to get the value

Reply:
May need some guidance on that. 
That said, in a future release of `screenlogicpy` I was thinking about moving those entities out of "config" and into "sensors" as I now know that data represents a live value, and not a configuration setting.

Reply:
TIL about EntityDescriptions. Interesting.